WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+	}
+
+	/**
+	 * Check for DOCTYPE declaration.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_doctype_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== stripos( $file_content, 'DOCTYPE' ) ) {
+			$sniff_helper['doctype'] = true;
+		}
+	}
+
+	/**
+	 * Check that at the very least the following two files exist: index.php
+	 * and style.css.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_minimum_file_check( $themedir ) {
+		global $sniff_helper;
+		if ( file_exists( $themedir . '/style.css' ) ) {
+			$sniff_helper['style_file_used'] = true;
+		}
+		if ( file_exists( $themedir . '/index.php' ) ) {
+			$sniff_helper['index_file_used'] = true;
+		}
+	}
+
+	/**
+	 * Check if readme.txt or readme.md exists.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_readme_file_check( $themedir ) {
+		global $sniff_helper;
+		if ( file_exists( $themedir . '/readme.txt' ) || file_exists( $themedir . '/readme.md' ) ) {
+			$sniff_helper['readme_file_used'] = true;
+		}
+	}
+
+	/**
+	 * Check for blacklisted files.
+	 * If found place in $sniff_helper array for later use
+	 *
+	 * @param array $themefiles contains the uri of files for the theme.
+	 */
+	public function blacklist_file_check( $themefiles ) {
+		global $sniff_helper;
+		$blacklist = array(
+				'thumbs.db'				=> 'Windows thumbnail store',
+				'desktop.ini'			=> 'windows system file',
+				'project.properties'	=> 'NetBeans Project File',
+				'project.xml'			=> 'NetBeans Project File',
+				'.kpf'					=> 'Komodo Project File',
+				'php.ini'				=> 'PHP server settings file',
+				'dwsync.xml'			=> 'Dreamweaver project file',
+				'error_log'				=> 'PHP error log',
+				'web.config'			=> 'Server settings file',
+				'.sql'					=> 'SQL dump file',
+				'__MACOSX'				=> 'OSX system file',
+				'.lubith'				=> 'Lubith theme generator file',
+				'.zip'					=> 'compressed file',
+
+			);
+		foreach ( $themefiles as $themefile ) {
+			foreach ( $blacklist as $key => $reason ) {
+				if ( false !== strpos( $themefile, $key ) ) {
+					$sniff_helper['files_not_allowed'][] = $key . ' - ' . $reason;
+				}
+			}
+			if ( false !== strpos( $themefile, '\.' ) ) {
+				if ( '\.' !== substr( $themefile, -2 ) && '\..' !== substr( $themefile, -3 ) ) {
+					$sniff_helper['files_not_allowed'][] = 'hidden file or folder:';
+					$sniff_helper['files_not_allowed'][] = $themefile;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Check if screenshot file exists.
+	 * Check screenhot max width 1200px.
+	 * Check screenshot max height 900px.
+	 * Check screenshot ratio 4:3.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_screenshot_checks( $themedir ) {
```

Comment:
This is not suited for a tokenized line sniff check. How do you chek a token for an image file? I think is is better done here, because it is accessing a file directly. It is also requiring to retrieve image properties. What if there was no screenshot at all? If you are committed to the Abstract_ThemeSniff class then let's take advantage of what it can do.